app/services/classroom_config.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class ClassroomConfig
+  CONFIGURABLES = %w[issues].freeze
+
+  attr_reader :github_repository
+
+  def initialize(github_repository)
+    raise ArgumentError, 'Invalid configuration repo' unless github_repository.branch_present? 'github-classroom'
+    @github_repository = github_repository
+  end
+
+  def setup_repository(repo)
+    configs_tree = @github_repository.branch_tree('github-classroom')
+    GitHub::Errors.with_error_handling do
+      configs_tree.tree.each do |config|
+        send("generate_#{config.path}", repo, config.sha) if CONFIGURABLES.include? config.path
+      end
+
+      repo.remove_branch('github-classroom')
+    end
+  end
+
+  def finished_setup?(repo)
+    repo.branch_present? 'github-classroom'
+  end
+
+  private
+
+  # Internal: Generates issues for the assignment_repository based on the configs
+  #
+  # repo - GitHubRepository for which to perform the configuration
+  #                   setups
+  # tree_sha     - sha of the 'issues' tree
+  #
+  # Returns nothing
+  def generate_issues(repo, tree_sha)
+    GitHub::Errors.with_error_handling do
```

Comment:
Same here. If a method we're using can raise a GitHub::Error, we can rescue that specifically instead of wrapping in this.